File: usecase/src/main/java/br/org/restaurantepopular/usecase/user/IUserRepository.java
package br.org.restaurantepopular.usecase.user;

import br.org.restaurantepopular.entity.User;

import java.util.Optional;
import java.util.List;

public interface IUserRepository {

    Optional<User> save(User user);

    void update(User user);

    List<User> listUser();

    Optional<User> findById(String id);

    void delete(String id);
}

File: usecase/src/main/java/br/org/restaurantepopular/usecase/user/IUserRepository.kt
package br.org.restaurantepopular.usecase.user

import br.org.restaurantepopular.entity.User

interface IUserRepository {

    fun save(user: User): User
    fun update(user: User)
    fun listUser(): List<User>
    fun findById(id: String): User
    fun delete(id: String)
}